### PR TITLE
Do not refresh token on every request and cache verified ones

### PR DESCRIPTION
**Pull request by cesarvarela**
For comparison:

https://staging-aiid.netlify.app/apps/submitted/

https://pr-3014--staging-aiid.netlify.app/apps/submitted/

### CODE REVIEW

File: site/gatsby-site/server/context.ts [hunk 1]
```diff
@@ -56,35 +51,75 @@ export const verifyToken = async (token: string) => {
...
         id: userId,
         roles: userData?.roles,
-    }
+    };
 }
 
-async function getUserFromHeader(header: string, client: MongoClient) {
+async function getTokenCache(token: string, client: MongoClient) {
+    const db = client.db('customData');
+    const collection = db.collection('tokenCache');
```
Comment (kepae) [inline site/gatsby-site/server/context.ts:66]: drive-by: let's discuss how to cache tokens before creating a DB collection for this.

**Comment by cesarvarela** [comment]
In favor of #3220
